mllib-local/src/main/scala/org/apache/spark/ml/linalg/Vectors.scala

```diff
@@ -239,6 +247,14 @@ object Vectors {
     }.toSeq)
   }
 
+  /**
+   * Calculate the dot product of two vectors.
+   *
+   * If `size` does not match an [[IllegalArgumentException]] is thrown.
+   */
+  @Since("3.0.0")
+  def dot(v1: Vector, v2: Vector): Double = BLAS.dot(v1, v2)
```

Comment:
Actually, do we need this method? BLAS.dot() already exists. I can see an instance method taking a single arg for parity with Pyspark, but this doesn't add much.

Reply:
It is private to spark, hence the simple wrapping: https://github.com/apache/spark/blob/master/mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala#L26

Reply:
Ah right, fair point. Still you can just call a.dot(b) after the first method is added, no?

Reply:
Yes indeed I can. Lemme fix.  Thanks!

Reply:
Oh, I meant, I don't think there is value in adding this method, because a _caller_ can use `a.dot(b)` directly.

Reply:
Ah, I got ya.  Yep, I can remove those.